src/doc/rustc/src/platform-support/nto-qnx.md

```diff
@@ -130,18 +141,35 @@ To compile for QNX Neutrino (aarch64 and x86_64) and Linux (x86_64):
 
 ```bash
 export build_env='
-    CC_aarch64-unknown-nto-qnx710=qcc
-    CFLAGS_aarch64-unknown-nto-qnx710=-Vgcc_ntoaarch64le_cxx
-    CXX_aarch64-unknown-nto-qnx710=qcc
+    CC_i586_pc_nto_qnx700=qcc
+    CFLAGS_i586_pc_nto_qnx700=-Vgcc_ntox86_cxx
+    CXX_i586_pc_nto_qnx700=qcc
+    AR_i586_pc_nto_qnx700=ntox86-ar
+
+    CC_aarch64_unknown_nto_qnx710=qcc
+    CFLAGS_aarch64_unknown_nto_qnx710=-Vgcc_ntoaarch64le_cxx
+    CXX_aarch64_unknown_nto_qnx710=qcc
     AR_aarch64_unknown_nto_qnx710=ntoaarch64-ar
-    CC_x86_64-pc-nto-qnx710=qcc
-    CFLAGS_x86_64-pc-nto-qnx710=-Vgcc_ntox86_64_cxx
-    CXX_x86_64-pc-nto-qnx710=qcc
-    AR_x86_64_pc_nto_qnx710=ntox86_64-ar'
+
+    CC_aarch64_unknown_nto_qnx710_iosock=qcc
+    CFLAGS_aarch64_unknown_nto_qnx710_iosock=-Vgcc_ntoaarch64le_cxx
+    CXX_aarch64_unknown_nto_qnx710_iosock=qcc
+    AR_aarch64_unknown_nto_qnx710_iosock=ntoaarch64-ar
+
+    CC_aarch64_unknown_nto_qnx700=qcc
+    CFLAGS_aarch64_unknown_nto_qnx700=-Vgcc_ntoaarch64le_cxx
+    CXX_aarch64_unknown_nto_qnx700=qcc
+    AR_aarch64_unknown_nto_qnx700=ntoaarch64-ar
+
+    CC_x86_64_pc_nto_qnx710=qcc
+    CFLAGS_x86_64_pc_nto_qnx710=-Vgcc_ntox86_64_cxx
+    CXX_x86_64_pc_nto_qnx710=qcc
+    AR_x86_64_pc_nto_qnx710=ntox86_64-ar
+    '
 
 env $build_env \
     ./x.py build \
-        --target aarch64-unknown-nto-qnx710,x86_64-pc-nto-qnx710,x86_64-unknown-linux-gnu \
+        --target aarch64-unknown-nto-qnx710_iosock,aarch64-unknown-nto-qnx710,x86_64-pc-nto-qnx710,x86_64-unknown-linux-gnu \
```

Comment:
The environment variables are given for:

* `i586-pc-nto-qnx700`
* `aarch64-unknown-nto-qnx710`
* `aarch64-unknown-nto-qnx710_iosock`
* `aarch64-unknown-nto-qnx700`
* `x86_64-pc-nto-qnx710`

The following targets are missing:

* `aarch64-unknown-nto-qnx800`
* `x86_64-pc_nto-qnx710_iosock`
* `x86_64-pc-nto-qnx800`

The build command builds:

* `aarch64-unknown-nto-qnx710_iosock`
* `aarch64-unknown-nto-qnx710`
* `x86_64-pc-nto-qnx710`
* `x86_64-unknown-linux-gnu

Because these lists don't match, and it's going to just produce a longer and longer build command, perhaps we should:

1. Add a section for each major QNX Neutrino version (7.0, 7.1, 8.0) 
2. Within that add a section for each supported target
3. In each section, explain the flags the `qcc` compiler requires.
4. Add one general section explaining how to set the environment variables (using the target name but with `_` instead of `-`) based on the information in the target specific sections, with perhaps a worked example for `aarch64-unknown-nto-qnx710`.

Then people can assemble their own build command for whichever of the (currently) eight QNX targets they want to use.

Reply:
Please review my latest change.

Reply:
Looks good. Thanks!